provider/cloudflare/cloudflare.go

```diff
@@ -339,7 +344,22 @@ func (p *CloudFlareProvider) submitChanges(ctx context.Context, changes []*cloud
 // AdjustEndpoints modifies the endpoints as needed by the specific provider
 func (p *CloudFlareProvider) AdjustEndpoints(endpoints []*endpoint.Endpoint) []*endpoint.Endpoint {
 	adjustedEndpoints := []*endpoint.Endpoint{}
+
+	zoneNameIDMapper, idToZoneMapper, err := p.prepareZoneMappings()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"endpoints": endpoints,
+		}).WithError(err).Errorf("Cannot adjust endpoints because zone mapping failed")
+
+		return endpoints
+	}
+
 	for _, e := range endpoints {
+		if err := p.addPlanInfoToEndpoint(e, zoneNameIDMapper, idToZoneMapper); err != nil {
+			log.WithFields(log.Fields{
+				"endpoint": e,
+			}).WithError(err).Errorf("Unable to attach Cloudflare plan information to endpoint")
+		}
```

Comment:
Are we okay ignoring the error here?

Reply:
The error here is provided for information purposes. If we reach that error, it would mean that we are trying to attach a record to a Cloudflare zone that is not registered or accessible to the Cloudflare token in use.

AFAIK, the current behavior of ExternalDNS doesn't break the running process, just report about, which is what we are doing here

Reply:
@Raffo Bump

Reply:
@Raffo Any updates?